app/v1/models/request/AddressModel.scala

```diff
@@ -56,24 +58,35 @@ object AddressModel {
     }
 
   }
+  private[models] def validateDateAsPriorDate(earlierDate: DateModel, optionalLaterDate: Option[DateModel]) = {
+    optionalLaterDate.fold(true){ laterDate =>
+      val earlierModelAsDate = LocalDate.parse(earlierDate.dateString, DateTimeFormatter.ofPattern("dd-MM-yyyy"))
+      val laterModelAsDate = LocalDate.parse(laterDate.dateString, DateTimeFormatter.ofPattern("dd-MM-yyyy"))
+      Logger.warn("[NameModel][validateDateAsPriorDate] The provided earlierDate is after the laterDate.")
+      earlierModelAsDate.isBefore(laterModelAsDate) || earlierModelAsDate.isEqual(laterModelAsDate)
+    }
+  }
 
   private def commonError(fieldName: String) = {
     JsonValidationError(s"There has been an error parsing the $fieldName field. Please check against the regex.")
   }
 
-  implicit val reads: Reads[AddressModel] = (
-    addressTypePath.readNullable[AddressType] and
-      line1Path.read[AddressLine] and
-      line2Path.readNullable[AddressLine] and
-      line3Path.readNullable[AddressLine] and
-      line4Path.readNullable[AddressLine] and
-      line5Path.readNullable[AddressLine] and
-      countryCodePath.read[String].flatMap(countryCode =>
-        postcodePath.readNullable[Postcode].filter(commonError("Post code"))(checkPostcodeMandated(_, countryCode))) and
-      countryCodePath.read[String] and
-      startDatePath.read[DateModel] and
-      endDatePath.readNullable[DateModel]
-    ) (AddressModel.apply _)
+  private def startDateAfterEndDateError = JsonValidationError("The given start date is after the given end date.")
+
+  implicit val reads: Reads[AddressModel] = for {
+    addressType <- addressTypePath.readNullable[AddressType]
+    line1 <- line1Path.read[AddressLine]
+    line2 <- line2Path.readNullable[AddressLine]
+    line3 <- line3Path.readNullable[AddressLine]
+    line4 <- line4Path.readNullable[AddressLine]
+    line5 <- line5Path.readNullable[AddressLine]
+    countryCode <- countryCodePath.read[String]
+    postcode <- postcodePath.readNullable[Postcode].filter(commonError("Post code"))(checkPostcodeMandated(_, countryCode))
+    startDate <- startDatePath.read[DateModel]
+    endDate <- endDatePath.readNullable[DateModel].filter(startDateAfterEndDateError)(validateDateAsPriorDate(startDate, _))
```

Comment:
There needs to be validation on endDate to ensure that this is not after the current date